7 - Metaprogramação/README.md

```diff
@@ -1,38 +1,97 @@
 # 7 - Metaprogramação
 
 ## Introdução
-* O que é metaprogramação
-* Propósitos
-  * Empacotamento
-  * Teste
-  * Gerenciamento de dependências
-* Exemplos
+
+De forma geral, programar seria algo como, "criar código para resolver problemas": Estamos basicamente criando programas a partir de códigos, que são executados para realizar tarefas, solucionando esses problemas. Mas e metaprogramação?
+
+Metaprogramação é um conjunto de programas voltados para solucionar problemas do *processo* de criar programas. São ferramentas que tem como objetivo testar, compilar e executar seu código de forma mais eficiente. Geralmente em programas pequenos não temos problemas com esse tipo de tarefa, mas conforme avançamos um sistema e aumentamos sua complexidade, elas passam a se tornar cada vez mais inviáveis de serem gerenciadas manualmente pelos programadores. Pense em metaprogramação em algo como "programas que operam em outros programas".
```

Comment:
Metaprogramação não é bem um conjunto de programas, né? A definição da wikipédia em inglês é boa: metaprogramação é uma técnica de programação na qual programas de computadores possuem a habilidade de manipular outros programas de computadores (os dados manipulados pelo programa são outros programas)." Original em inglês: "Metaprogramming is a programming technique in which [computer programs](https://en.wikipedia.org/wiki/Computer_program) have the ability to treat other programs as their data"

Reply:
Ajustei um pouco o texto. Ficou melhor?